core/src/main/scala/anorm/Column.scala

```diff
@@ -748,10 +751,17 @@ sealed trait JavaTimeColumn {
    *     as(SqlParser.scalar[LocalDateTime].single)
    * }}}
    */
-  implicit val columnToLocalDateTime: Column[LocalDateTime] =
-    temporalColumn[LocalDateTime]({ ts: Long =>
+  implicit val columnToLocalDateTime: Column[LocalDateTime] = nonNull { (value, meta) =>
+
+    def millisToLocalDateTime(ts: Long) =
       LocalDateTime.ofInstant(Instant.ofEpochMilli(ts), ZoneId.systemDefault)
-    }, "Java8 LocalDateTime")
+
+    value match {
+      case localDateTime: LocalDateTime => Right(localDateTime)
```

Comment:
How a `LocalDateTime` could be there in JDBC ?

Reply:
https://dev.mysql.com/doc/relnotes/connector-j/8.0/en/news-8-0-23.html

From the important changes:

> Also, with the implementation of the new mechanism, a getObject(columnIndex) call on a DATETIME column returns a LocalDateTime object now instead of a String. To receive a String like before, use getObject(columnIndex, String.class) instead. 

I haven't read any specs, but I assume getObject(int) is free to return any object.

Reply:
I guess you could freely interpret this to allow for that:

> This method may also be used to read database-specific abstract data types.

\- https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSet.html#getObject-int-